src/data/mv_polynomial.lean

```diff
@@ -739,6 +750,24 @@ by rw [vars, degrees_C, multiset.to_finset_zero]
 @[simp] lemma vars_X (h : 0 ≠ (1 : α)) : (X n : mv_polynomial σ α).vars = {n} :=
 by rw [X, vars_monomial h.symm, finsupp.support_single_ne_zero zero_ne_one.symm]
 
+lemma mem_support_not_mem_vars_zero {f : mv_polynomial σ α} {x : σ →₀ ℕ} (H : x ∈ f.support) {v : σ} (h : v ∉ vars f) :
+  x v = 0 :=
+begin
+  rw [vars, multiset.mem_to_finset] at h,
+  rw ←not_mem_support_iff,
+  by_contradiction,
+  have : v ∈ degrees f := begin
```

Comment:
Why did you change this to a nested `begin ... end` block?
The version with braces is perfectly fine.

Reply:
Oh right, I had changed that earlier and didn't notice when commiting. Should I change it back?

Reply:
ie. are braces preferred over nested `begin ... end` blocks?

Reply:
Yes, braces are preferred over nest `begin ... end` blocks. (Extracting a preliminary lemma is also often a reasonable option if you really want to mark it out as a separate argument.)

Reply:
Ok. I will change that now.